src/main/java/AntidoteBackend.java

```diff
@@ -0,0 +1,116 @@
+            // Do something like `erl -sname whatever` to check and get it running if not
+            // KEEP THIS
+            myOtpNode = new OtpNode(NodeName);
+            myOtpNode.setCookie(cookie);
+            myOtpMbox = myOtpNode.createMbox();
+            myOtpMbox.registerName(MailBox);
+        } catch (IOException e) {
+            e.printStackTrace();
+            throw new RuntimeException(e);
+        }
+    }
+
+    public abstract OtpErlangBinary value(OtpErlangBinary JavaObjectId) throws NoSuchObjectException;
+
+    public abstract OtpErlangBinary update(OtpErlangBinary JavaObjectId, OtpErlangBinary binary)
+            throws NoSuchObjectException;
+
+    // If we want to do any analysis on the operation based on current state before
+    // it becomes a valid operation.
+    public abstract OtpErlangBinary downstream(OtpErlangBinary JavaObjectId, OtpErlangBinary binary)
+            throws NoSuchObjectException;
+
+    public abstract OtpErlangTuple snapshot(OtpErlangBinary JavaObjectId) throws NoSuchObjectException;
+
+    // For when erlang wants to instantiate a new erlang object for a to be created
+    // java object and needs a corresponding id.
+    // Must be 20 bytes or things will go poorly
+    public abstract OtpErlangBinary newJavaObjectId();
+
+    public void run() {
+        System.out.println(myOtpNode);
+        while (true) {
+            try {
+                OtpErlangTuple tuple = (OtpErlangTuple) myOtpMbox.receive();
+
+                last_pid = (OtpErlangPid) tuple.elementAt(0);
+
+                OtpErlangTuple payload = (OtpErlangTuple) tuple.elementAt(1);
+
+                // Antidote sends the JavaId, operation, and binary args
+                OtpErlangBinary JavaObjectId = (OtpErlangBinary) payload.elementAt(0);
+                int status_enum = ((OtpErlangLong) payload.elementAt(1)).intValue();
+                OtpErlangBinary binary = (OtpErlangBinary) payload.elementAt(2);
+
+                switch (Status.values()[status_enum]) {
+                    case read:
+                        myOtpMbox.send(last_pid, value(JavaObjectId));
+                        break;
+
+                    case update:
+                        myOtpMbox.send(last_pid, update(JavaObjectId, binary));
+                        break;
+
+                    case snapshot:
+                        myOtpMbox.send(last_pid, snapshot(JavaObjectId));
+                        break;
+
+                    case downstream:
+                        myOtpMbox.send(last_pid, downstream(JavaObjectId, binary));
+                        break;
+
+                    case newjavaid:
+                        myOtpMbox.send(last_pid, newJavaObjectId());
+                        break;
+                }
+            } catch (NoSuchObjectException e) {
+                OtpErlangAtom atom = new OtpErlangAtom("getobject");
+                myOtpMbox.send(last_pid, atom);
+            } catch (Exception e) {
```

Comment:
The point of this catch is to deal with any errors that could be raised by some object implementation. If the only thing that someone could raise in their counter/set implementation is a Runtime exception then sure. I've written it as a catch-all so that the antidote side will throw an error instead of merely timing out and continuing when the backend errors.

Reply:
Actually, turns out in changing something else, this was hiding some exceptions from Jinterface. So it is no a runtime exception catch for this case.

Reply:
This is really a difficult issue --- in fact Andrew has a recent paper on it (https://dl.acm.org/doi/abs/10.1145/2980983.2908086).   The trouble is that you have two sources of exceptions: anticipated (or potentially anticipated) exceptions thrown by your code, and exceptions thrown by client code that you call.  You'd like to be able to catch the ones your code throws in a different block from catching the ones the client code throws.  Sadly there's not a great way to handle this.  The best one I know of is to make sure you use a set of custom exception classes for your own exceptions --- `BackendCheckedException` and `BackendUncheckedException` or something --- and just be sure that any exceptions you generate are either handled at the generation site, already inherited from one of your exception classes, or caught and wrapped in one of your classes before being re-thrown.  If you do this, you can then have a `catch(Exception)` block that you're reasonably confident only applied to client exceptions.  But a comment to that effect would go a long way!